The job **`x86_64-gnu-llvm-18`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/12254391755/job/34185208926) [(plain)](https://github.com/rust-lang/rust/commit/23cc55f48e036bfa5f9ef0388633c477fdc6a291/checks/34185208926/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
#16 exporting to docker image format
#16 sending tarball 27.2s done
#16 DONE 40.3s
##[endgroup]
Setting extra environment values for docker:  --env ENABLE_GCC_CODEGEN=1 --env GCC_EXEC_PREFIX=/usr/lib/gcc/
[CI_JOB_NAME=x86_64-gnu-llvm-18]
debug: `DISABLE_CI_RUSTC_IF_INCOMPATIBLE` configured.
---
sccache: Starting the server...
##[group]Configure the build
configure: processing command line
configure: 
configure: build.configure-args := ['--build=x86_64-unknown-linux-gnu', '--llvm-root=/usr/lib/llvm-18', '--enable-llvm-link-shared', '--set', 'rust.randomize-layout=true', '--set', 'rust.thin-lto-import-instr-limit=10', '--enable-verbose-configure', '--enable-sccache', '--disable-manage-submodules', '--enable-locked-deps', '--enable-cargo-native-static', '--set', 'rust.codegen-units-std=1', '--set', 'dist.compression-profile=balanced', '--dist-compression-formats=xz', '--set', 'rust.lld=false', '--disable-dist-src', '--release-channel=nightly', '--enable-debug-assertions', '--enable-overflow-checks', '--enable-llvm-assertions', '--set', 'rust.verify-llvm-ir', '--set', 'rust.codegen-backends=llvm,cranelift,gcc', '--set', 'llvm.static-libstdcpp', '--enable-new-symbol-mangling']
configure: target.x86_64-unknown-linux-gnu.llvm-config := /usr/lib/llvm-18/bin/llvm-config
configure: llvm.link-shared     := True
configure: rust.randomize-layout := True
configure: rust.thin-lto-import-instr-limit := 10
---
failures:

---- [codegen] tests/codegen/intrinsics/offset.rs stdout ----

error: verification with 'FileCheck' failed
status: exit status: 1
command: "/usr/lib/llvm-18/bin/FileCheck" "--input-file" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/codegen/intrinsics/offset/offset.ll" "/checkout/tests/codegen/intrinsics/offset.rs" "--check-prefix=CHECK" "--check-prefix" "NONMSVC" "--allow-unused-prefixes" "--dump-input-context" "100"
--- stderr -------------------------------
/checkout/tests/codegen/intrinsics/offset.rs:21:12: error: CHECK: expected string not found in input
/checkout/tests/codegen/intrinsics/offset.rs:21:12: error: CHECK: expected string not found in input
 // CHECK: %[[R:.*]] = getelementptr inbounds i32, ptr %p, [[SIZE]] %d
           ^
/checkout/obj/build/x86_64-unknown-linux-gnu/test/codegen/intrinsics/offset/offset.ll:13:65: note: scanning from here
define noundef ptr @offset_isize(ptr noundef %p, i64 noundef %d) unnamed_addr #0 {
                                                                ^
/checkout/obj/build/x86_64-unknown-linux-gnu/test/codegen/intrinsics/offset/offset.ll:13:65: note: with "SIZE" equal to "i64"
define noundef ptr @offset_isize(ptr noundef %p, i64 noundef %d) unnamed_addr #0 {
/checkout/obj/build/x86_64-unknown-linux-gnu/test/codegen/intrinsics/offset/offset.ll:16:2: note: possible intended match here
/checkout/obj/build/x86_64-unknown-linux-gnu/test/codegen/intrinsics/offset/offset.ll:16:2: note: possible intended match here
 %_0 = getelementptr inbounds i8, ptr %p, i64 %0
/checkout/tests/codegen/intrinsics/offset.rs:30:12: error: CHECK: expected string not found in input
/checkout/tests/codegen/intrinsics/offset.rs:30:12: error: CHECK: expected string not found in input
 // CHECK: %[[R:.*]] = getelementptr inbounds i64, ptr %p, [[SIZE]] %d
           ^
/checkout/obj/build/x86_64-unknown-linux-gnu/test/codegen/intrinsics/offset/offset.ll:21:65: note: scanning from here
define noundef ptr @offset_usize(ptr noundef %p, i64 noundef %d) unnamed_addr #0 {
                                                                ^
/checkout/obj/build/x86_64-unknown-linux-gnu/test/codegen/intrinsics/offset/offset.ll:21:65: note: with "SIZE" equal to "i64"
define noundef ptr @offset_usize(ptr noundef %p, i64 noundef %d) unnamed_addr #0 {
/checkout/obj/build/x86_64-unknown-linux-gnu/test/codegen/intrinsics/offset/offset.ll:24:2: note: possible intended match here
/checkout/obj/build/x86_64-unknown-linux-gnu/test/codegen/intrinsics/offset/offset.ll:24:2: note: possible intended match here
 %_0 = getelementptr inbounds i8, ptr %p, i64 %0

Input file: /checkout/obj/build/x86_64-unknown-linux-gnu/test/codegen/intrinsics/offset/offset.ll
Check file: /checkout/tests/codegen/intrinsics/offset.rs


-dump-input=help explains the following input dump.
Input was:
<<<<<<
<<<<<<
            1: ; ModuleID = 'offset.6eb952c938ba959d-cgu.0' 
            2: source_filename = "offset.6eb952c938ba959d-cgu.0" 
            3: target datalayout = "e-m:e-p270:32:32-p271:32:32-p272:64:64-i64:64-i128:128-f80:128-n8:16:32:64-S128" 
            4: target triple = "x86_64-unknown-linux-gnu" 
            5:  
            6: ; Function Attrs: nonlazybind uwtable 
            7: define noundef ptr @offset_zst(ptr noundef %p, i64 noundef %d) unnamed_addr #0 { 
            9:  ret ptr %p 
           10: } 
           11:  
           11:  
           12: ; Function Attrs: nonlazybind uwtable 
           13: define noundef ptr @offset_isize(ptr noundef %p, i64 noundef %d) unnamed_addr #0 { 
check:21'0                                                                     X~~~~~~~~~~~~~~~~~~ error: no match found
check:21'1                                                                                         with "SIZE" equal to "i64"
check:21'0     ~~~~~~~
check:21'0     ~~~~~~~
           15:  %0 = mul nuw i64 %d, 4 
check:21'0     ~~~~~~~~~~~~~~~~~~~~~~~~
           16:  %_0 = getelementptr inbounds i8, ptr %p, i64 %0 
check:21'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
check:21'2      ?                                                possible intended match
           17:  ret ptr %_0 
check:21'0     ~~~~~~~~~~~~~
           18: } 
check:21'0     ~~
check:21'0     ~
check:21'0     ~
           20: ; Function Attrs: nonlazybind uwtable 
check:21'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           21: define noundef ptr @offset_usize(ptr noundef %p, i64 noundef %d) unnamed_addr #0 { 
check:21'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
check:30'0                                                                     X~~~~~~~~~~~~~~~~~~ error: no match found
check:30'1                                                                                         with "SIZE" equal to "i64"
check:30'0     ~~~~~~~
check:30'0     ~~~~~~~
           23:  %0 = mul nuw i64 %d, 8 
check:30'0     ~~~~~~~~~~~~~~~~~~~~~~~~
           24:  %_0 = getelementptr inbounds i8, ptr %p, i64 %0 
check:30'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
check:30'2      ?                                                possible intended match
           25:  ret ptr %_0 
check:30'0     ~~~~~~~~~~~~~
           26: } 
check:30'0     ~~
check:30'0     ~
check:30'0     ~
           28: attributes #0 = { nonlazybind uwtable "probe-stack"="inline-asm" "target-cpu"="x86-64" } 
check:30'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
check:30'0     ~
check:30'0     ~
           30: !llvm.module.flags = !{!0, !1} 
check:30'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           31: !llvm.ident = !{!2} 
check:30'0     ~~~~~~~~~~~~~~~~~~~~
check:30'0     ~
check:30'0     ~
           33: !0 = !{i32 8, !"PIC Level", i32 2} 
check:30'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           34: !1 = !{i32 2, !"RtLibUseGOT", i32 1} 
check:30'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           35: !2 = !{!"rustc version 1.85.0-nightly (bf89f5a48 2024-12-10)"} 
check:30'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
------------------------------------------


---- [codegen] tests/codegen/ptr-arithmetic.rs stdout ----
---- [codegen] tests/codegen/ptr-arithmetic.rs stdout ----

error: verification with 'FileCheck' failed
status: exit status: 1
command: "/usr/lib/llvm-18/bin/FileCheck" "--input-file" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/codegen/ptr-arithmetic/ptr-arithmetic.ll" "/checkout/tests/codegen/ptr-arithmetic.rs" "--check-prefix=CHECK" "--check-prefix" "NONMSVC" "--allow-unused-prefixes" "--dump-input-context" "100"
--- stderr -------------------------------
/checkout/tests/codegen/ptr-arithmetic.rs:9:12: error: CHECK: expected string not found in input
/checkout/tests/codegen/ptr-arithmetic.rs:9:12: error: CHECK: expected string not found in input
 // CHECK: %[[TEMP:.+]] = getelementptr inbounds i32, ptr %p, [[WORD]] %n
/checkout/obj/build/x86_64-unknown-linux-gnu/test/codegen/ptr-arithmetic/ptr-arithmetic.ll:7:69: note: scanning from here
/checkout/obj/build/x86_64-unknown-linux-gnu/test/codegen/ptr-arithmetic/ptr-arithmetic.ll:7:69: note: scanning from here
define noundef ptr @i32_add(ptr noundef readnone %p, i64 noundef %n) unnamed_addr #0 {
                                                                    ^
/checkout/obj/build/x86_64-unknown-linux-gnu/test/codegen/ptr-arithmetic/ptr-arithmetic.ll:7:69: note: with "WORD" equal to "i64"
define noundef ptr @i32_add(ptr noundef readnone %p, i64 noundef %n) unnamed_addr #0 {
/checkout/obj/build/x86_64-unknown-linux-gnu/test/codegen/ptr-arithmetic/ptr-arithmetic.ll:10:2: note: possible intended match here
/checkout/obj/build/x86_64-unknown-linux-gnu/test/codegen/ptr-arithmetic/ptr-arithmetic.ll:10:2: note: possible intended match here
 %_0 = getelementptr inbounds i8, ptr %p, i64 %0
/checkout/tests/codegen/ptr-arithmetic.rs:20:12: error: CHECK: expected string not found in input
/checkout/tests/codegen/ptr-arithmetic.rs:20:12: error: CHECK: expected string not found in input
 // CHECK: %[[DELTA:.+]] = sub nsw [[WORD]] 0, %n
/checkout/obj/build/x86_64-unknown-linux-gnu/test/codegen/ptr-arithmetic/ptr-arithmetic.ll:15:69: note: scanning from here
/checkout/obj/build/x86_64-unknown-linux-gnu/test/codegen/ptr-arithmetic/ptr-arithmetic.ll:15:69: note: scanning from here
define noundef ptr @i32_sub(ptr noundef readnone %p, i64 noundef %n) unnamed_addr #0 {
                                                                    ^
/checkout/obj/build/x86_64-unknown-linux-gnu/test/codegen/ptr-arithmetic/ptr-arithmetic.ll:15:69: note: with "WORD" equal to "i64"
define noundef ptr @i32_sub(ptr noundef readnone %p, i64 noundef %n) unnamed_addr #0 {
/checkout/obj/build/x86_64-unknown-linux-gnu/test/codegen/ptr-arithmetic/ptr-arithmetic.ll:17:2: note: possible intended match here
/checkout/obj/build/x86_64-unknown-linux-gnu/test/codegen/ptr-arithmetic/ptr-arithmetic.ll:17:2: note: possible intended match here
 %.neg = mul i64 %n, -4
/checkout/tests/codegen/ptr-arithmetic.rs:30:12: error: CHECK: expected string not found in input
/checkout/tests/codegen/ptr-arithmetic.rs:30:12: error: CHECK: expected string not found in input
 // CHECK: %[[TEMP:.+]] = getelementptr inbounds i32, ptr %p, [[WORD]] %d
/checkout/obj/build/x86_64-unknown-linux-gnu/test/codegen/ptr-arithmetic/ptr-arithmetic.ll:23:72: note: scanning from here
/checkout/obj/build/x86_64-unknown-linux-gnu/test/codegen/ptr-arithmetic/ptr-arithmetic.ll:23:72: note: scanning from here
define noundef ptr @i32_offset(ptr noundef readnone %p, i64 noundef %d) unnamed_addr #0 {
                                                                       ^
/checkout/obj/build/x86_64-unknown-linux-gnu/test/codegen/ptr-arithmetic/ptr-arithmetic.ll:23:72: note: with "WORD" equal to "i64"
define noundef ptr @i32_offset(ptr noundef readnone %p, i64 noundef %d) unnamed_addr #0 {
/checkout/obj/build/x86_64-unknown-linux-gnu/test/codegen/ptr-arithmetic/ptr-arithmetic.ll:26:2: note: possible intended match here
/checkout/obj/build/x86_64-unknown-linux-gnu/test/codegen/ptr-arithmetic/ptr-arithmetic.ll:26:2: note: possible intended match here
 %_0 = getelementptr inbounds i8, ptr %p, i64 %0

Input file: /checkout/obj/build/x86_64-unknown-linux-gnu/test/codegen/ptr-arithmetic/ptr-arithmetic.ll
Check file: /checkout/tests/codegen/ptr-arithmetic.rs


-dump-input=help explains the following input dump.
Input was:
<<<<<<
<<<<<<
            1: ; ModuleID = 'ptr_arithmetic.bab9a0ec4242e570-cgu.0' 
            2: source_filename = "ptr_arithmetic.bab9a0ec4242e570-cgu.0" 
            3: target datalayout = "e-m:e-p270:32:32-p271:32:32-p272:64:64-i64:64-i128:128-f80:128-n8:16:32:64-S128" 
            4: target triple = "x86_64-unknown-linux-gnu" 
            5:  
            6: ; Function Attrs: mustprogress nofree norecurse nosync nounwind nonlazybind willreturn memory(none) uwtable 
            7: define noundef ptr @i32_add(ptr noundef readnone %p, i64 noundef %n) unnamed_addr #0 { 
check:9'0                                                                          X~~~~~~~~~~~~~~~~~~ error: no match found
check:9'1                                                                                              with "WORD" equal to "i64"
check:9'0      ~~~~~~~
check:9'0      ~~~~~~~
            9:  %0 = shl nuw i64 %n, 2 
check:9'0      ~~~~~~~~~~~~~~~~~~~~~~~~
           10:  %_0 = getelementptr inbounds i8, ptr %p, i64 %0 
check:9'0      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
check:9'2       ?                                                possible intended match
           11:  ret ptr %_0 
check:9'0      ~~~~~~~~~~~~~
           12: } 
check:9'0      ~~
check:9'0      ~
check:9'0      ~
           14: ; Function Attrs: mustprogress nofree norecurse nosync nounwind nonlazybind willreturn memory(none) uwtable 
check:9'0      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           15: define noundef ptr @i32_sub(ptr noundef readnone %p, i64 noundef %n) unnamed_addr #0 { 
check:9'0      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
check:20'0                                                                         X~~~~~~~~~~~~~~~~~~ error: no match found
check:20'1                                                                                             with "WORD" equal to "i64"
check:20'0     ~~~~~~~
check:20'0     ~~~~~~~
           17:  %.neg = mul i64 %n, -4 
check:20'0     ~~~~~~~~~~~~~~~~~~~~~~~~
check:20'2      ?                       possible intended match
           18:  %_0 = getelementptr inbounds i8, ptr %p, i64 %.neg 
check:20'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           19:  ret ptr %_0 
check:20'0     ~~~~~~~~~~~~~
           20: } 
check:20'0     ~~
check:20'0     ~
check:20'0     ~
           22: ; Function Attrs: mustprogress nofree norecurse nosync nounwind nonlazybind willreturn memory(none) uwtable 
check:20'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           23: define noundef ptr @i32_offset(ptr noundef readnone %p, i64 noundef %d) unnamed_addr #0 { 
check:20'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
check:30'0                                                                            X~~~~~~~~~~~~~~~~~~ error: no match found
check:30'1                                                                                                with "WORD" equal to "i64"
check:30'0     ~~~~~~~
check:30'0     ~~~~~~~
           25:  %0 = shl nuw i64 %d, 2 
check:30'0     ~~~~~~~~~~~~~~~~~~~~~~~~
           26:  %_0 = getelementptr inbounds i8, ptr %p, i64 %0 
check:30'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
check:30'2      ?                                                possible intended match
           27:  ret ptr %_0 
check:30'0     ~~~~~~~~~~~~~
           28: } 
check:30'0     ~~
check:30'0     ~
check:30'0     ~
           30: attributes #0 = { mustprogress nofree norecurse nosync nounwind nonlazybind willreturn memory(none) uwtable "probe-stack"="inline-asm" "target-cpu"="x86-64" } 
check:30'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
check:30'0     ~
check:30'0     ~
           32: !llvm.module.flags = !{!0, !1} 
check:30'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           33: !llvm.ident = !{!2} 
check:30'0     ~~~~~~~~~~~~~~~~~~~~
check:30'0     ~
check:30'0     ~
           35: !0 = !{i32 8, !"PIC Level", i32 2} 
check:30'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           36: !1 = !{i32 2, !"RtLibUseGOT", i32 1} 
check:30'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           37: !2 = !{!"rustc version 1.85.0-nightly (bf89f5a48 2024-12-10)"} 
check:30'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
------------------------------------------



```

</details>

--- a/compiler/rustc_codegen_ssa/src/mir/place.rs
+++ b/compiler/rustc_codegen_ssa/src/mir/place.rs
@@ -422,10 +422,7 @@ impl<'a, 'tcx, V: CodegenObject> PlaceRef<'tcx, V> {
             layout.size
         };
 
-        let llval = bx.inbounds_gep(bx.cx().backend_type(self.layout), self.val.llval, &[
-            bx.cx().const_usize(0),
-            llindex,
-        ]);
+        let llval = bx.inbounds_gep(bx.cx().backend_type(layout), self.val.llval, &[llindex]);
         let align = self.val.align.restrict_for_offset(offset);
         PlaceValue::new_sized(llval, align).with_type(layout)
     }

--- a/tests/codegen/bounds-checking/gep-issue-133979.rs
+++ b/tests/codegen/bounds-checking/gep-issue-133979.rs
@@ -1,0 +1,22 @@
+//! Issue: <https://github.com/rust-lang/rust/issues/133979>
+//! Check that bounds checking are eliminated.
+
+//@ compile-flags: -Copt-level=2
+
+#![crate_type = "lib"]
+
+// CHECK-LABEL: @test(
+#[no_mangle]
+fn test(a: &[&[u8]]) -> u32 {
+    // CHECK-NOT: panic_bounds_check
+    a.iter()
+        .enumerate()
+        .map(|(y, b)| {
+            b.iter()
+                .enumerate()
+                .filter(|(_, c)| **c == b'A')
+                .map(|(x, _)| a[y][x] as u32)
+                .sum::<u32>()
+        })
+        .sum()
+}

--- a/tests/codegen/gep-index.rs
+++ b/tests/codegen/gep-index.rs
@@ -1,0 +1,37 @@
+//! Check that index and offset use the same getelementptr format.
+
+//@ revisions: NO-OPT OPT
+//@[NO-OPT] compile-flags: -Copt-level=0
+//@[OPT] compile-flags: -Copt-level=1
+
+#![crate_type = "lib"]
+
+struct Foo(i32, i32);
+
+// CHECK-LABEL: @index_on_struct(
+#[no_mangle]
+fn index_on_struct(a: &[Foo], index: usize) -> &Foo {
+    // CHECK: getelementptr inbounds %Foo, ptr %a.0, {{i64|i32}} %index
+    &a[index]
+}
+
+// CHECK-LABEL: @offset_on_struct(
+#[no_mangle]
+fn offset_on_struct(a: *const Foo, index: usize) -> *const Foo {
+    // CHECK: getelementptr inbounds %Foo, ptr %a, {{i64|i32}} %index
+    unsafe { a.add(index) }
+}
+
+// CHECK-LABEL: @index_on_i32(
+#[no_mangle]
+fn index_on_i32(a: &[i32], index: usize) -> &i32 {
+    // CHECK: getelementptr inbounds i32, ptr %a.0, {{i64|i32}} %index
+    &a[index]
+}
+
+// CHECK-LABEL: @offset_on_i32(
+#[no_mangle]
+fn offset_on_i32(a: *const i32, index: usize) -> *const i32 {
+    // CHECK: getelementptr inbounds i32, ptr %a, {{i64|i32}} %index
+    unsafe { a.add(index) }
+}

